File: client/src/app/models/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  cod: number;
  weight: number;
  pictureUrl: string;
  type: string;
  brand: string;
  quantity: number;
  date: string;
  isWeighed: boolean;
}
export interface ProductParams {
  orderBy: string;
  searchTerm?: string;
  types: string[];
  brands: string[];
  pageNumber: number;
  pageSize: number;
}

File: API/Entities/Product.cs

namespace API.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public double COD { get; set; }
        public double Weight { get; set; }
        public string PictureUrl { get; set; }
        public string Type { get; set; }
        public string Brand { get; set; }
        public int Quantity { get; set; }
        public bool isWeighed { get; set; }
        public DateTime Date { get; set; } = DateTime.UtcNow;
        public string PublicId { get; set; }
    }
}